blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpEventsQueryHandler.java

```diff
@@ -27,35 +28,46 @@ public HttpEventsQueryHandler(EventsIO searchIO) {
+            String responseBody = null;
+            final Timer.Context httpEventsFetchTimerContext = httpEventsFetchTimer.time();
+            try {
+                HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
+                Map<String, List<String>> params = requestWithParams.getQueryParams();
+
+                if (params == null || params.size() == 0) {
+                    throw new InvalidDataException("Query should contain at least one query parameter");
+                }
 
-            parseDateFieldInQuery(params, "from");
-            parseDateFieldInQuery(params, "until");
-            List<Map<String, Object>> searchResult = searchIO.search(tenantId, params);
-            responseBody = objectMapper.writeValueAsString(searchResult);
-        } catch (InvalidDataException e) {
-            log.error(String.format("Exception %s", e.toString()));
-            responseBody = String.format("Error: %s", e.getMessage());
-            status = HttpResponseStatus.BAD_REQUEST;
-        } catch (Exception e) {
-            log.error(String.format("Exception %s", e.toString()));
-            responseBody = String.format("Error: %s", e.getMessage());
-            status = HttpResponseStatus.INTERNAL_SERVER_ERROR;
-        } finally {
-            DefaultHandler.sendResponse(ctx, request, responseBody, status);
-            httpEventsFetchTimerContext.stop();
+                parseDateFieldInQuery(params, "from");
+                parseDateFieldInQuery(params, "until");
+                List<Map<String, Object>> searchResult = searchIO.search(tenantId, params);
+                responseBody = objectMapper.writeValueAsString(searchResult);
+            } catch (InvalidDataException e) {
+                log.error(String.format("Exception %s", e.toString()));
+                responseBody = String.format("Error: %s", e.getMessage());
+                status = HttpResponseStatus.BAD_REQUEST;
+            } catch (Exception e) {
+                log.error(String.format("Exception %s", e.toString()));
+                responseBody = String.format("Error: %s", e.getMessage());
+                status = HttpResponseStatus.INTERNAL_SERVER_ERROR;
+            } finally {
+                DefaultHandler.sendResponse(ctx, request, responseBody, status);
+                httpEventsFetchTimerContext.stop();
+            }
+        }
+        else if(request.getMethod().toString().equalsIgnoreCase("OPTIONS")){
+            HttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+            response.setHeader("Access-Control-Allow-Origin", "*");
+            response.setHeader("Access-Control-Allow-Methods", "GET");
+            response.setHeader("Access-Control-Allow-Headers", "X-Auth-Token, Accept");
+            response.setHeader("Access-Control-Max-Age", "1728000");
+            HttpResponder.respond(ctx, request, response);
         }
```

Comment:
Do we need a final else case here to handle where the request is neither GET or OPTIONS?